### ISSUE TITLE
How to Install

### ISSUE THREAD

**Issue by grac-zh** [Ximi1970/Manually-Sort-Folders#5]
as many others, i want my manually sorted folders back (after updating to thunderbird 60). downloading your pre-version  doesn't work, because firefox prevents it, saying that the file is corrupt.
when will the final new version of "manually sort folders" plugin be available for not-specialised users?
thanks.

**Comment by Ximi1970** [comment]
I think you are trying to install it in Firefox...

This is the way to do it:

Download the .xpi file from the release tab on the main Manually Sort Folders page

https://github.com/Ximi1970/Manually-Sort-Folders/releases

and right click on the .xpi file and select "Save Link as".

or right click below and select "Save Link as":

https://github.com/Ximi1970/Manually-Sort-Folders/releases/download/v2.0.0pre1/manually_sort_folders_2.0.0pre1.xpi

Startup Thunderbird and go to the menu Tools -> Add-ons
Click on the button next to the search bar (left side) in the Add-ons Manager page.
You get a new menu. Select "Install Add-on from file.
Select the file and click the "Open" button.
A new dialog will appear. Wait a few seconds for the countdown and click "Install".
Restart Thunderbird.

Ready.

You can tweak some of the actions in the preferences:

For TB < 60: Go to the Add-on page again and click the preferences button in the Manually Sort Folders add-on.
For TB 60: Go to the menu Tools -> Add-on preferences and select Manually Sort Folders

Set all the options you want to use and press ok.

Edit: Update for v2.0.0pre1, TB 68

**Comment by Gluups** [comment]
That version works quite nice on Windows 10.
Feedback from other systems will be welcome, so that the version can be properly deployed to the official platforms.
Version 60.3.0 of Thunderbird has recently been deployed by automatic update, so if the deployment of the extension is not done we are going to get many tickets asking how to get the good version.

**Comment by grac-zh** [comment]
no, i didn't want to install your plugin in firefox, but in thunderbird - of course!
but going to github works only with firefox.
and still, firefox tells me (with your second link):

The add-on downloaded from this site could not be installed because it appears to be corrupt.

(see also the screenshot attached).

the first link doesn't show a xpi file.

i use firefox 63.0.1. (64bit) on a windows 10 64bit machine.

stephan


![2018-11-14_143739](https://user-images.githubusercontent.com/15222117/48486871-f3188980-e81c-11e8-83ac-da1740157ef1.png)

**Comment by Gluups** [comment]
Well, if Firefox tells you that, it is normal, as it is a Thunderbird extension.
As Ximi explained you, you must not click on the link, with the main button of the mouse, but right click on the link to get a contextual menu, and then click on download.
Then in Thunderbird, in the addon manager tab, click on the button on top, you get a menu, click on "install from a file". Then you select the xpi file you downloaded from Firefox.

**Comment by grac-zh** [comment]
well, finally i found the xpi file here:

https://github.com/Ximi1970/Manually-Sort-Folders/releases/

what a confusion with these links ...

now it's downloaded and installed in thunderbird - the result is as it was before, thanks.

**Comment by Ximi1970** [comment]
@Gluups, Jonathan (protz) promised to merge the pull request I made a few weeks ago.
https://github.com/protz/Manually-Sort-Folders/pull/72#issuecomment-433485806
Hopefully he will find some time to update the Thunderbird add-on store.

**Comment by schneeschmelze** [comment]
I've just installed the fix in TB 60.3.1 on macOS 10.13.6, and it seems to work well. Thanks a lot, I'm quite grateful for your effort!

**Comment by ghost** [comment]
v2.0.0pre1 can be installed by dragging, but the account field is left blank. Therefore, it cannot be used. The close button does not move either.
"extensions.tbsortfolders@xulforum.org.accounts
"extensions.tbsortfolders@xulforum.org.defaultaccount"
The account field remains blank after reset.

Please tell me how the account appears correctly in the account field of this add-on and how the restart and close buttons work.

Thunderbird 68 64bit  Windows10pro 64bit 1903

**Comment by despecial** [comment]
@pmboom18 try this:
Go to Thunderbird>>Tools>>Options>>Advanced>>General>>Config Editor
Scroll down to extensions.tbsortfolders@xulforum.org.accounts
Right-click and select Reset
Repeat for extensions.tbsortfolders@xulforum.org.defaultaccount
Restart Thunderbird

**Comment by ghost** [comment]
despecial 　
thank you
Unfortunately, as shown in the image, the list items under the "Sort acounts" tab are left blank.

I recreated my profile and tried a new installation of Thunderbird 68.
Then add "v 2.0. 0 pre1" 

Unfortunately this too. The list field remains blank.
The "restart button" and "close button" also don't work at all.

I also tried something like https://github.com/protz/Manually-Sort-Folders/issues/82 but it is left blank

Does v2.0.0pre1 fully support Thunderbird 68?

![not work](https://user-images.githubusercontent.com/53461014/64740471-6ddf2d80-d530-11e9-8916-0cd7d9c00a31.jpg)

**Comment by despecial** [comment]
so the previous tip doesnt work anymore? @pmboom18

**Comment by ghost** [comment]
Dear despecial,

I tried a hint with Thunderbird68 and Manual-Sort-Folders v2.0.0pre1, but it didn't work. The account field remains blank.

The "restart button" and "close button" also don't work at all.

**Comment by BNGI** [comment]
Well, it worked... kind of with TB68.0. Kind of, because you could not move Local Folders to the top of the list (I tried and I couldn't). Anyway, yesterday I upgraded TB to 68.1 and I got the same problem as pmboom18. Tried despecial suggestion, but to no avail; tried removing the add-on and reinstall it and it leads to the same problem (attached the english option panel for manually sort folders):

![T-Bird Manually Sort Folders](https://user-images.githubusercontent.com/55240829/64784368-fb724b80-d537-11e9-9669-eee7612f05f7.png)

**Comment by ghost** [comment]
Looking at the error console, it looks like this.

WebExtensions: Loading packed extension from C:\Users\<username>\AppData\Roaming\Thunderbird\Profiles\xxxxxxxx.default\extensions\tbsortfolders@xulforum.org.xpi

WebExtensions: Loading add-on preferences from  C:\Users\<username>\AppData\Roaming\Thunderbird\Profiles\xxxxxxxx.default\extensions\tbsortfolders@xulforum.org.xpi ext-legacy.js:142
WebExtensions: Firing profile-after-change listeners for tbsortfolders@xulforum.org

Thunderbird 68 seems to recognize "Manually-Sort-Folders v 2.0. 0 pre1" as a legacy add-on, but does Thunderbird 68 support it ?

**Comment by kotaro366** [comment]
As with you, the list items under the "Account Sort" tab are left blank.
The “Close button” and “Restart button” also do not work.
I hope it is available.

Windows10pro 64bit 1903　Thunderbird 68.10 64bit

**Comment by Shitennouji** [comment]
As pmboom18 and BNGI report, the situation is exactly the same.
Issues are also being reported at the [MozillaZine.jp Forum in Japan.](https://forums.mozillazine.jp/viewtopic.php?f=16&t=17732&p=64564#p64564)

**Comment by amuellerde** [comment]
Hi, here the same.
TB68.1
error console shows errrors:

* tbsortfolders.Logging.LogMod.Formatter is undefined logging.jsm:92
    setupLogging resource://tbsortfolders/logging.jsm:92
    init resource://tbsortfolders/logging.jsm:62
    <anonym> resource://tbsortfolders/logging.jsm:149
    <anonym> chrome://tbsortfolders/content/folderPane.js:9
    <anonym> chrome://tbsortfolders/content/folderPane.js:192
    loadScript resource:///modules/Overlays.jsm:485
    load resource:///modules/Overlays.jsm:203
    load resource:///modules/Overlays.jsm:40
    registerNonBootstrapped chrome://messenger/content/parent/ext-legacy.js:185
    forEach self-hosted:266
    registerNonBootstrapped chrome://messenger/content/parent/ext-legacy.js:181
    AsyncFunctionNext self-hosted:839
    fetchOverlay resource:///modules/Overlays.jsm:447
    load resource:///modules/Overlays.jsm:93
    load resource:///modules/Overlays.jsm:40
    registerNonBootstrapped chrome://messenger/content/parent/ext-legacy.js:185
    forEach self-hosted:266
    registerNonBootstrapped chrome://messenger/content/parent/ext-legacy.js:181
    AsyncFunctionNext self-hosted:839
* TypeError: tbsortfolders.Logging.LogMod.Formatter is undefined logging.jsm:92:7
    setupLogging resource://tbsortfolders/logging.jsm:92
    init resource://tbsortfolders/logging.jsm:62
    <anonym> resource://tbsortfolders/logging.jsm:149
    <anonym> chrome://tbsortfolders/content/ui.js:5
* ReferenceError: can't access lexical declaration `tblog' before initialization ui.js:241:3
    on_load chrome://tbsortfolders/content/ui.js:241
    onload chrome://tbsortfolders/content/tbsortfolders.xul:1

**Comment by protz** [comment]
@amuellerde I hit all the errors you described and fixed them -- see https://jonathan.protzenko.fr/manually-sort-folders/manually-sort-folders-201909221547.xpi for an updated version which hopefully should be approved shortly

Thanks,

Jonathan

**Comment by ghost** [comment]
https://jonathan.protzenko.fr/manually-sort-folders/manually-sort-folders-201909221547.xpi

f you select a startup folder on the Add tab, even if you select a custom folder, only No Folder is displayed in the right column and the list of accounts dropdown is not displayed.

             
![2019](https://user-images.githubusercontent.com/53461014/65395471-3dbc4800-ddd6-11e9-8645-aa4c831c351e.jpg)

**Comment by kotaro366** [comment]
Hi protz.
 manually-sort-folders-201909221547.xpi . But there is still a bug.

When you select a "custom folder" with the left button on the "Extra settings" tab page, clicking the right button only displays "No Folder Selection" and you cannot select another folder.

**Comment by protz** [comment]
Yes, this one has been separately reported. I need to look into it -- I hadn't tested this feature yesterday.

**Comment by BNGI** [comment]
@protz I'm still having the problem that on the "Sort accounts" tab, I don't have "Local Folders" in the pane where you list the account that can be sorted. As such, I cannot sort Local to make Local Folders appear first (see screenshot).

Thank you for your help.

![TB Manually Sort Folders](https://user-images.githubusercontent.com/55240829/65426259-d86c5500-dddd-11e9-994a-419c7e194da1.png)

**Comment by protz** [comment]
all: please try https://jonathan.protzenko.fr/manually-sort-folders/manually-sort-folders-201909230901.xpi for the fix to the startup folder

@BNGI : I mentioned this in another thread but what my addon does is merely flipping a pref for the default account which in the past used to result in this account appearing first -- can you try picking the local folders as the default account via Tools > Account Settings, using Account Actions (bottom left) > Set as Default? If this no longer has the side-effect of putting it first like it used to, then this is something that changed in Thunderbird, over which I have no control -- your best bet is probably to hit bugzilla.mozilla.org and file a bug, CCing :protz (me) there so that I can remain looped into the discussion.

**Comment by BNGI** [comment]
@protz Hello Jonathan; thank you for your quick reply.

The option you mentioned is now greyed out in Thunderbird so you cannot set Local Folders as default account anymore. Probably, this has the effect that Local Folder is now the last of the accounts shown with no option to making it first. I think I'll file a bug on bugzilla and see what happens (letting you know); I can't believe they missed this one...

Thank you,

**Comment by ghost** [comment]
When selecting a startup folder on the Add tab, the account drop-down list in the right column is now displayed when a custom folder is selected. However, even if "Open folder at start" is set, there is a bug that the set folder does not open but the "last opened folder" opens.

Especially after the second start.

**Comment by kotaro366** [comment]
I looked at the pmboom18 writing and confirmed it with my Thunderbird 68.1.

Like pmboom 18, the folder setting at startup does not seem to start in the set folder.

"201909230926.xpi" at https://jonathan.protzenko.fr/manually-sort-folders/ had the same problem.

It's a shame that my technology can only confirm the operation. I'm sorry.

**Comment by kotaro366** [comment]
Improved startup folder settings

Once the set folder can be fixed,

Many people will be happy as in ver 60.

I hope it will be improved soon.

**Comment by ghost** [comment]
https://jonathan.protzenko.fr/manually-sort-folders/manually-sort-folders-201909230901.xpi does not remember the manually configured start folder.
The folder that was last opened will open and start up.
Please correct it.
